snuba/datasets/entity.py

```diff
@@ -28,20 +28,20 @@ class Entity(Describable, ABC):
     def __init__(
         self,
         *,
-        storages: Sequence[Storage],
+        storages: Sequence[StorageAndMappers],
```

Comment:
I think this is not modelling the reality quite accurately. If we think about entities and storages as nodes, each entry in this list is supposed to be an edge

```
Entity -- (mappers, is_writeable) --> Storage
```

But the way this is represented in code is not quite that. We have the `writeable_storage` field which is also connected in the same way but is given special treatment while still existing in the `storages` array. 

I think `StorageAndMappers` should become something more general like `EntityStorageConnection` which contains the mappers, is_writeable, and in the future whatever else we decide to add to that relationship.

The yaml is already specifying things that way, why not make the code do so as well? That way you won't need the special case that you have in the entity builder

https://github.com/getsentry/snuba/pull/3591/files#diff-530c557e763d4aa8baeb0c82eb931da0d8e7206d886d3b266357330ed9506145R145

Reply:
This way you can also remove that reverse lookup I was talking about in the last PR:

https://github.com/getsentry/snuba/pull/3545#discussion_r1054774522

Reply:
I think the improvement to make configuration and code more consistent with each other makes sense. 

So if I understand correctly, is what are suggesting to remove `writable_storage` as an attribute on `Entity` and change `entity.get_writable_storage()` to get a storage from `entity.storages: Sequence[EntityStorageConnection]`?

Reply:
exactly